src/test/java/org/folio/rest/api/RequestBatchAPITest.java

```diff
@@ -0,0 +1,192 @@
+package org.folio.rest.api;
+
+import static org.folio.rest.api.RequestsApiTest.requestStorageUrl;
+import static org.folio.rest.api.StorageTestSuite.TENANT_ID;
+import static org.folio.rest.api.StorageTestSuite.storageUrl;
+import static org.folio.rest.support.ResponseHandler.empty;
+import static org.folio.rest.support.ResponseHandler.json;
+import static org.folio.rest.support.builders.RequestRequestBuilder.OPEN_AWAITING_PICKUP;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import java.net.URL;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.folio.rest.support.ApiTests;
+import org.folio.rest.support.JsonResponse;
+import org.folio.rest.support.Response;
+import org.folio.rest.support.builders.RequestRequestBuilder;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+public class RequestBatchAPITest extends ApiTests {
+
+  @Before
+  public void beforeEach() throws Exception {
+    StorageTestSuite.deleteAll(requestStorageUrl());
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs("request");
+  }
+
+  @Test
+  public void canUpdateRequestPositionsInBatch() throws Exception {
+    UUID itemId = UUID.randomUUID();
+
+    // 1st create some sample requests for an item
+    JsonObject firstRequest = createRequestForItemAtPosition(itemId, 1);
+    JsonObject secondRequest = createRequestForItemAtPosition(itemId, 2);
+
+    // 2nd: Swap positions for these requests
+    JsonObject firstRequestWithNoPosition = firstRequest.copy();
+    JsonObject secondRequestWithNoPosition = secondRequest.copy();
+
+    // 2.1 - release previous positions
+    secondRequestWithNoPosition.remove("position");
+    firstRequestWithNoPosition.remove("position");
+
+    // 2.2 - set new positions
+    firstRequest.put("position", 2);
+    secondRequest.put("position", 1);
+
+    CompletableFuture<Response> putCompleted = new CompletableFuture<>();
+    JsonObject allRequests = buildBatchUpdateRequest(firstRequestWithNoPosition,
+      secondRequestWithNoPosition, firstRequest, secondRequest);
+
+    client.put(batchRequestStorageUrl(), allRequests, TENANT_ID, empty(putCompleted));
+
+    Response response = putCompleted.get(5, TimeUnit.SECONDS);
+    assertThat(response.getStatusCode(), is(204));
+
+    // 3rd: Retrieve requests from DB and assert changes.
```

Comment:
What information does this comment add over the code below it?

Reply:
I've refactored tests, it is outdated now.